src/test/java/de/tum/cit/aet/artemis/programming/icl/LocalCIIntegrationTest.java

```diff
@@ -67,6 +69,7 @@
 import de.tum.cit.aet.artemis.programming.util.LocalRepository;
 
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@Execution(ExecutionMode.SAME_THREAD)
```

Comment:
Can you please explain in a comment why both annotations are necessary? Be as verbose as possible!